### PR TITLE
ref(mediators): Convert the other external request mediators 

### PR DESCRIPTION
**Pull request by Christinarlong**
[x] SelectRequester
[x] IssueLinkRequester
[x] tests
[x] typing

loggers

### CODE REVIEW

File: src/sentry/sentry_apps/external_requests/issue_link_requester.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                        "project": self.group.project.slug,
+                        "group": self.group.id,
+                        "uri": self.uri,
+                        "error_message": str(e),
+                    },
+                )
+
+            if not self._validate_response(response):
+                raise APIError(
+                    f"Invalid response format from sentry app {self.sentry_app} when linking issue"
```
Comment (markstory) [inline src/sentry/sentry_apps/external_requests/issue_link_requester.py:92]: :clap: Nice improvement on the error message.

File: tests/sentry/sentry_apps/external_requests/test_select_requester.py [hunk 2]
```diff
@@ -93,13 +91,11 @@ def test_invalid_response_missing_value(self):
         )
 
         with pytest.raises(APIError):
-            SelectRequester.run(
+            SelectRequester(
                 install=self.install,
                 project_slug=self.project.slug,
-                group=self.group,
```
Comment (markstory) [inline tests/sentry/sentry_apps/external_requests/test_select_requester.py:99]: Nice! Dataclasses helped find some mistakes. :clap:

**Review by markstory — APPROVED** [review]

File: src/sentry/sentry_apps/external_requests/issue_link_requester.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    install: RpcSentryAppInstallation
+    uri: str
+    group: Group
+    fields: dict[str, Any]
+    user: RpcUser
+    action: str
+
+    def run(self) -> dict[str, Any]:
+        with transaction.atomic(router.db_for_write(Group)):
```
Comment (Christinarlong) [inline src/sentry/sentry_apps/external_requests/issue_link_requester.py:61]: I dont think we need this transaction ? It doesn't look like we're making an db requests here or downstream unless the callers of this are ? but in that case wouldn't that be the caller's responsibility to wrap their call in a transaction?

Comment (markstory) [reply src/sentry/sentry_apps/external_requests/issue_link_requester.py:61]: Yeah, I'm not seeing and DB operations in this or the called methods. We could skip the transaction here.

**Review by markstory — APPROVED** [review]

**Pull request merged**